.github/workflows/build-and-test.yml

```diff
@@ -60,11 +63,17 @@ jobs:
           mkdir -p ./pr
           echo "event_name=$EVENT_NAME,pr_number=$PR_NUMBER,sha=$SHA," > ./pr/pr_number
           cd ./packages/dapp && zip -r pull-request.zip ./dist && cp pull-request.zip ../../pr
+
       - uses: actions/upload-artifact@v3
         with:
           name: pr
           path: pr/
 
+      - name: Spell Check
+        run: |
+          LIST="$(find . -type f \( -name '*.sol' -o -name '*.tsx' -o -name '*.ts' \) ! -path '*/node_modules/*' ! -path '*/lib/*' ! -path '*/types/*')"
+          yes | npx cspell $LIST
+
```

Comment:
`spell-check.yml` should run all the time anyways so I don't see the purpose of you consolidating it inside of `build-and-test.yml`